code/clue.js

```diff
@@ -1,59 +1,218 @@
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
 
-const rooms = []
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+const mystery = {
+  killer: "",
+  weapons: "",
+  room: "",
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//STEP 3 - Functions
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  mystery.killer = randomSelector(suspects);
+
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+
+  document.getElementById("killerName").innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById("killerAge").innerHTML = `${mystery.killer.age}`;
+  document.getElementById("killerImage").src = `${mystery.killer.image}`;
+  document.getElementById("killerOccupation").innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById("killerDescription").innerHTML = `${mystery.killer.description}`;
+  document.getElementById("killerCard").onclick = pickKiller;
+};
+
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+
+  document.getElementById("weaponName").innerHTML = `${mystery.weapon.name} ${mystery.weapon.material}`;
+  document.getElementById("weaponImage").src = `${mystery.weapon.image}`;
+  document.getElementById("weaponCard").onclick = pickWeapon;
+};
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+  document.getElementById("roomName").innerHTML = `${mystery.room}`;
+  document.getElementById("roomCard").onclick = pickRoom;
+};
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// STEP 4 
+const revealMystery = () => {
+  document.getElementById("mystery").innerHTML = `The murder was comitted by ${mystery.killer.firstName} in the ${mystery.room} with a ${mystery.weapon.name}`;
+  document.getElementById("mysteryBtn").onclick = revealMystery;
```

Comment:
What happens if you click the button and not all cards are picked? You could fix that by adding some conditional statements to this function to show one message if not all cards are picked, and only show this "The murder was committed..."-message after all cards are picked :)